### PR TITLE
[FL-2797] Signal Generator app

### PR DESCRIPTION
**Pull request by nminaylov**
# What's new

- Signal generator app prototype, currently supports: 
  - Generating PWM using TIM1 or LPTIM2 outputs
  - Configurable clock output to MCO pin
- furi_hal_pwm
- furi_hal_clock: MCO configuration

# Verification 

- Build and upload plugins
- Check Tools/Signal Generator

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: firmware/targets/f7/furi_hal/furi_hal_pwm.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#endif
+
+#include <stdint.h>
+
+typedef enum {
+    FuriHalPwmOutputIdTim1PA7,
+    FuriHalPwmOutputIdLptim2PA4,
+} FuriHalPwmOutputId;
+
+void furi_hal_pwm_start(FuriHalPwmOutputId channel, uint32_t freq, uint8_t duty);
```
Comment (DrZlo13) [inline firmware/targets/f7/furi_hal/furi_hal_pwm.h:24]: Documentation

**Review by DrZlo13 — CHANGES_REQUESTED** [review]

File: firmware/targets/f7/furi_hal/furi_hal_pwm.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#pragma once
```
Comment (DrZlo13) [inline firmware/targets/f7/furi_hal/furi_hal_pwm.h:5]:
```
/**
 * @file furi_hal_pwm.h
 * PWM control HAL
 */
 ```

**Review by skotopes — APPROVED** [review]

**Review by DrZlo13 — APPROVED** [review]

**Comment by fuznutz73** [comment]
Any ideas on how to open the frequency possibilities below 1Hz (ie. 14.035 Hz, same as the NEW opticon receivers)?

**Comment by skotopes** [comment]
@fuznutz73 use TIM2 to generate signal

**Pull request merged**